api/client/swarm/opts.go

```diff
@@ -99,27 +101,31 @@ func (o *AutoAcceptOption) String() string {
 }
 
 // Set sets a new value on this option
-func (o *AutoAcceptOption) Set(value string) error {
-	value = strings.ToUpper(value)
-	switch value {
-	case "", "NONE":
-		if accept, ok := o.values[WORKER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", WORKER)
-		}
-		if accept, ok := o.values[MANAGER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", MANAGER)
-		}
-		o.values[WORKER] = false
-		o.values[MANAGER] = false
-	case WORKER, MANAGER:
-		if accept, ok := o.values[value]; ok && !accept {
-			return fmt.Errorf("value NONE is incompatible with %s", value)
+func (o *AutoAcceptOption) Set(acceptValues string) error {
+	for _, value := range strings.Split(acceptValues, ",") {
+		value = strings.ToUpper(value)
+		switch value {
+		case "", NONE:
+			// NONE must stand alone, so if any non-NONE setting exists, conflict
+			if !o.values[NONE] && len(o.values) > 0 {
+				var s []string
+				for r := range o.values {
+					s = append(s, r)
+				}
+				return fmt.Errorf("value NONE is incompatible with %s", strings.Join(s, ", "))
+			}
+			o.values[WORKER] = false
+			o.values[MANAGER] = false
```

Comment:
Actually there shouldn't be a need to remove these items anyway, because this is unreachable if either one was set to true. Also, removing them would interfere with moving the compatibility check out of the loop, as I suggested in https://github.com/docker/docker/pull/24146/files#r70158372.

I would still highly recommend making the type of `o.values` `map[string]struct{}`, though.